src/ol/style/iconstyle.js

```diff
@@ -456,8 +456,8 @@ ol.style.IconImage_.get = function(image, src, size, crossOrigin, imageState) {
  */
 ol.style.IconImage_.prototype.determineTainting_ = function() {
   var context = ol.dom.createCanvasContext2D(1, 1);
-  context.drawImage(this.image_, 0, 0);
   try {
+    context.drawImage(this.image_, 0, 0);
```

Comment:
The only explanation I have is that if the canvas was tainted already, it would fail. But with the code you added in the constructor, this should not be the case.